src/main/java/org/mitre/synthea/engine/Logic.java

```diff
@@ -367,9 +371,35 @@ public boolean test(Person person, long time) {
   public static class ActiveCondition extends ActiveLogic {
     @Override
     public boolean test(Person person, long time) {
+      if(this.value_set != null){
+
+        Set<String> presentCodes = person.record.present.keySet();
+
+        List<String> valueSetCodes = Terminology.sess.getAllCodes(this.value_set)
+                .stream()
+                .map(code -> {return code.code;})
+                .collect(Collectors.toList());
+
+
+        // Takes intersection of codes in value set and codes present in the record and updates in place.
+        // RetainAll returns true if it changes something, you could use that fact to define a boolean
+        // if you wanted to check if ALL the codes in the value set are present.
+        presentCodes.retainAll(valueSetCodes);
```

Comment:
Does this impact the `person.record.present` map in any way? I.e., if codes are removed in this operation, do they also get removed in the `present` map?

Reply:
Yes I checked and it does, I'll change it to make an actual copy so it doesn't affect the original.